invenio/modules/search/searchext/engines/native.py

```diff
@@ -48,7 +48,7 @@ def search(self, user_info=None, collection=None):
     from invenio.modules.search.walkers.search_unit import SearchUnit
     results = get_results_cache(self._query, collection)
     if results is None:
-        results = self.query.accept(SearchUnit())
+        results = self.enhanced_query.accept(SearchUnit())
```

Comment:
But get_results_cache is native-indexer specific, isn't it? If this is moved to the api then what will the search() function from searchext/engines/native.py do?